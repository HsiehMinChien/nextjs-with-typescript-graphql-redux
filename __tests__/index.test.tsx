import { shallow } from "enzyme";
import React from "react";

import Clock from "../components/clock";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

describe("With Enzyme", () => {
  const initialState = { lastUpdate: 0, light: false };
  const mockStore = configureStore();
  let store;

  it("Clock component first test", () => {
    store = mockStore(initialState);
    const clock = shallow(
      <Provider store={store}>
        <Clock />
      </Provider>
    );
    expect(clock.props().children).toEqual(<Clock />);
    expect(clock).toMatchSnapshot();
  });
});
