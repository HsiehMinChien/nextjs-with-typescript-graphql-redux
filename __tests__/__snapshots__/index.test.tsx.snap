// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Enzyme Clock component first test 1`] = `ShallowWrapper {}`;
